0.38 [unused]

fabaff opened the pull request:
Feels free to add content here if you think that your contribution is worth its own section in the release notes. Thanks

### Custom state-card UI
A nice new feature is the possibility to create [custom state-card][custom-ui] in the frondend. Go ahead and write your own state-card for [lights][light], sensors, locks, etc.

### MQTT discovery
The latest MQTT feature is [discovery][mqtt-discovery] support. Do not confuse this which the Home Assistant [Discovery][discovery]. MQTT Discover is for MQTT-only. Similar to the HTTP sensor/binary sensor allows this to use devices with minimal configuration.

### Alert component
If you left your front door open, then the [`alert`][alert] component can be used to remind you of this by sending you repeating notifications at an intervals.

### All changes
#### New platforms/components
Will be created on Saturday

#### Improvements

dito

### Breaking changes
- The support for [LG webOS Smart TVs][webostv] was improved. This requires you to move `$HOME/.pylgtv` to `$HASS_CONFIG_DIR/webostv.conf` or Home Assistant will need to be paired with the TV again.

### If you need help...
...don't hesitate to use our [Forum](https://community.home-assistant.io/) or join us for a little [chat](https://gitter.im/home-assistant/home-assistant). The release notes have comments enabled but it's preferred if you use the former communication channels. Thanks.

### Reporting Issues
Experiencing issues introduced by this release? Please report them in our [issue tracker](https://github.com/home-assistant/home-assistant/issues). Make sure to fill in all fields of the issue template.

[webostv]: https://home-assistant.io/components/media_player.webostv/
[mqtt-discovery]: https://home-assistant.io/components/mqtt/#discovery
[discovery]: https://home-assistant.io/components/discovery/
[alert]: https://home-assistant.io/components/alert/
[light]: https://home-assistant.io/cookbook/custom_ui_by_andrey-git
[custom-ui]: https://home-assistant.io/developers/frontend_creating_custom_ui/

balloob committed:
Version bump to 0.38.0.dev0

fabaff committed:
Upgrade rpi-rf to 0.9.6 (#5611)

DavidLP committed:
Fixes: Pilight Switch rejects alphanumeric IDs #5119 (#5601)

kylehendricks committed:
Fixes the AsusWRT `ip neigh` regex to handle the possible IPv6 "router" flag (#5605)

See the last line here: http://linux-ip.net/gl/ip-cref/ip-cref-node62.html

MartinHjelmare committed:
Fix mysensors RGB and W light turn on (#5608)

* RGBW light needs a white value defined.
* Log error if V_RGB is not 6 characters and V_RGBW not 8 characters.

pvizeli committed:
Move part of image_processing tests (#5634)

* Move part of image_processing tests

* fix lint

pvizeli committed:
Aiohttp client unittest (#5635)

* add test for cleanup

* add test for mjpeg stream

balloob committed:
Prioritize core.js (#5636)

* Prioritize loading app core script

* change app skeleton to be small

balloob committed:
Update frontend (#5637)

balloob committed:
Update frontend (#5638)

* Ensure fonts get loaded quicker

* Update frontend

balloob committed:
Update frontend (#5640)

balloob committed:
Update frontend (#5642)

pvizeli committed:
Add unittest for cleanup not validate ssl stuff. (#5643)

fabaff committed:
Use device_state_attributes() for platforms (#5649)

* Use device_state_attributes() for platforms

* Update test

* Fix lint issue

balloob committed:
Update frontend (#5652)

* Return empty result when history date is in future

* Update frontend

fabaff committed:
Upgrade slacker to 0.9.40 (#5650)

mnoorenberghe committed:
`yarn setup_js_dev` was deleted (#5639)

Danielhiversen committed:
Add set_options_service for input select (#5630)

Danielhiversen committed:
bug fix in hue (#5623)

miniconfig committed:
Added new sensor component to monitor OpenEVSE chargers equipped with the WiFi Kit.

emlove committed:
[docker] cec install path fixed upstream (#5651)

tchellomello committed:
Fixes issue #5627 by bumping external Amcrest module to version 1.1.4 (#5662)

Add an optional extended description…

rytilahti committed:
Cleanup modes & available, bump version requirement (#5606)

* Cleanup modes & available, bump version requirement

* check for Noneness on available

rytilahti committed:
new yeelight backend lib, new features (#5296)

* initial yeelight based on python-yeelight

* adapt yeelight's discovery code & suppress exceptions on set_default

* Support flash & code cleanups

Adds simple pulse for flashing, needs to be refined.
This commit also includes changing transition from seconds to milliseconds,
and cleans up the code quite a bit.

* cleanup code, adjust default transition to 350

* bump required version to 0.0.13

* Cleaning up and marking todos, ready to be reviewed

* Renamed back to yeelight.
* Removed effect support for now until we have some sane effects available.
* Add "breath" notification for flash, currently hidden behind a False check due to unknown issue not accepting it.
* TODO/open points are marked as such.

* Fix a typo in rgb calculation

* yeelight_<bulbtype>_<mac> for autodetected bulbs

hostname from mdns seems to vary

* Lint fixes, add music mode, fix flash

* Flash transforms now to red and back
* Fix lint warnings
* Add initial music mode.

* remove unused mode logging, move set_mode to turn_on

* Add save_on_change configuration variable

* yeelight: check if music mode is on before enabling it.

* Fix linting, bump required python-yeelight version

* More linting fixes, use import when needed instead of saving the module handle

* Use OR instead of + for features assignment

* Fix color temperature support, convert non-rgb values to rgb values in rgb()

* Fix typo on duration, thanks @qzapwy for noticing

* yeelight: fix issues from review, behave when not available

* Implement available()
* Fix transition to take seconds instead of milliseconds
* Fix default configuration for detected bulbs
* Cache values fetched in update()
* Add return values for methods

* yeelight: kwarg-given transition overrides config, slight cleanups

* change settings back to optional, request update when calling add_devices

* As future version of python-yeelight will wrap exceptions, we can handle broken connections more nicely.

* bump yeelight library version

* Remove unused import

* set the default only when settings are changed and not, e.g., when turned on by automation

* update comment & fix linting

Danielhiversen committed:
Vlc (#5665)

* vlc default name

balloob committed:
Update frontend (#5669)

pvizeli committed:
Add unittests for FFmpeg and spliting binary sensor (#5659)

* Spliting ffmpeg binary sensor and move service to component.

* unittests for component

* add unittest for binary_sensor

* exclude camera for tests

adrianlzt committed:
Voice command API.AI. First import (#5462)

* Voice command API.AI. First import

* Fixes suggested by hound

* Fixing comments

* Fix pylint and pydocstyle errors

* Change how speech is defined

Also clean some unused constants, remove card type (not used), define
a message when action is not defined and improve the message when
action is unknown.

* Change how speech is defined

Clean some constants.
Improve error messages.
Delete card type, not used.

* Tests for new Api.ai component

* Use async_add_job to python compatibility. New test to measure response time

* Add async_action option to choose between waiting or not for the action to execute

* Travis-ci needs more time

* Removed timeout tests

* Removed timeout tests

* Added apiai to .coveragerc as specified by PR doc

michaelarnauts committed:
Add devices detected by ping as SOURCE_TYPE_ROUTER instead of GPS (#5625)

aequitas committed:
Load mqtt_eventstream before any other component. (#5629)

This ensures that initial state changes for components are broadcasted on the MQTT bus.

aequitas committed:
Rflink 433Mhz gateway platform and components (#4547)

* Initial sketches of rflink component.

* Add requirement.

* Properly load configuration.

* Bump rflink for graceful parse errors and protocol callback.

* Cleanup, documentation and linting.

* More documentation, first sensor implementation (temp & hum).

* Add brightness/dim support for newkaku protocol.

* Use separate class for dimmables.

* Make sure non-dimmable newkaku devices are turned on.

* Move some code around, add switches. Support loading from config.

* Fix bug in ignoring devices.

* Fix initial state assumption.

* Improve reliability on invalid conditions.

* Allow configuration of group for new devices.

* Sensor icons.

* Fix parsing negative numbers.

* Correct icon.

* Allow sending commands serial.

* Pluralize.

* Allow adding sensors from config.

* Fix ignoring devices and bugs in previous commit.

* Share know devices so devices from configuration don't get added as lights.

* Lookup unit from value_key.

* Remove debug.

* Start implementing event protocol in place of packet protocol.

- Added first test suite for sensors.
- This currently breaks light and switch.

* Refactor switch component to fit new rflink changes. Add test suite.

* Fix style.

* Refactor and test lights. Bring coverage to 100%.

* Use non-broken and production tested rflink module.

* Update requirements.

* Bump for logging.

* Improve readability.

* Do not use global variable but keep known device state in intended place.

* Improve docs.

* Make icon support generic.

* Disable overriding icons in config, as it belongs in customization. Only keep custom icon for entities that are able to detect a icon based on the thing they represent (sensors in this case).

* Implement configuration schema, overall refactor of magic values.

* Fix bug in config/test wait_for_ack.

* Small refactors.

* Move command logic into separate class.

* Convert command sending logic to class based pattern instead of using the event bus.

* Start not using bus for rflink event propagation to platforms.

* Do not use event bus for all entity types.

* Fire an event on the bus for every switch incoming rflink command.

* Resolve lint errors, remove some old code.

* Known devices no longer need to be registered separately.

* Log bus events.

* Event callback is a..... callback.

* Use full entity id for events.

* Move event sending to entity.

* Log incoming events.

* Make firing events optional inline with rfxtrx.

* Add foundation for signal repetition.

* Add signal repetition config and tests.

* Make plain switchable type explicitly configurable.

* Enable default entity settings for automatically added entities as well.

* Prevent default configuration leaking accross entities.

* Make sure device defaults don't get overwritten by defaults further down.

* Don't let fast state switching and repetitions turn your house into a disco.

* Make repetitions more responsive.

* Disable on/off fallback on dimmables as it currently doesn't play nice with repetitions.

* Use rflink that allows send_command_ack to be safely cancelled.

* Reduce duplication and make repeat work for non-ack.

* Implement reconnection logic.

* Improve reconnection logic.

* Also cancel repetitions when entity state is changed due to external command.

* Update requirements.

* Fix linting.

* Fix spelling.

* Don't lie.

* Fix lint.

* Support for automatically creating protocol translation (fixes spaces in device names).

* Returned support for dimmable and on/off entity.

* Duplicate code to fix linting issues with inheritance.

* Allow overriding unit of measurement from config.

mathewpeterson committed:
Refactors script/setup_docker_prereqs (#5506)

* Refactors script/setup_docker_prereqs

Refactors script/setup_docker_prereqs to allow toggling of packages
to being installed

* Adds support for openalpr to Docker

* Updates Dockerfile

Comments ENV directives in order to preserve cache.

* Fixes incorrect position of echo

* Fixes telldus installer by updating apt before pkg install

molobrakos committed:
Upgraded tellduslive (#5664)

michaelarnauts committed:
Move docker installation scripts to virtualization/Docker path. Splits out openalpr to seperate script. (#5676)

balloob committed:
Update frontend (#5683)

aequitas committed:
Only set LANG=C.UTF8 during install. (#5648)

balloob committed:
Update frontend (#5685)

aequitas committed:
Use same pattern for device defaults in both platforms. (#5682)

pvizeli committed:
Bugfix sonos group coordinator (#5691)

* Bugfix sonos group coordinator

* Fix tests

pvizeli committed:
Bugfix async blocking loop with xml parser. (#5694)

jabesq committed:
Hotfix for netatmo cameras (#5644)

* Fix for missing netatmo tags in 0.37

Also fix issue with SSL certificate for vpn_url

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* Netatmo welcome: vpn_url can be empty

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* add config floag to disable SSL verification for vpn_url

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* Import CONF_VERIFY_SSL from const

balloob committed:
Update frontend (#5696)

Danielhiversen committed:
up rfxtrx lib (#5687)

pvizeli committed:
Fix bug for UNREACH devices / Variable handling and update. (#5689)

* Fix bug for UNREACH devices / Variable handling and update.

* fix track_time

* update after data after creation

* add message output

* change unreach

* change unreach code

* Revert "change unreach code"

This reverts commit f58430de3cb854d5f6f886bfe4d59899c1efadbf.

* update pyhomematic

pvizeli committed:
Bugfix sonso source input (#5699)

turbokongen committed:
[lock.zwave] Bugfix Zwave lock  (#5619)

* Bugfix state

* remove debug

aequitas committed:
Prevent infinite loop in crossconfigured mqtt event streams (#5624)

* Prevent events about MQTT messages received to cause infinite loop when two HA instances are crossconfigured for mqtt_eventstream.

* Fix linting

* Publish all MQTT received events except incoming from eventstream. Also make it configurable.

kellerza committed:
[sensor/sma] handle units correctly (#5657)

tboyce021 committed:
Fix hue lightgroups not syncing state (#5702)

pvizeli committed:
Cleanup climate platform for async update_ha_state / migrate generic thermostat (#5679)

* Cleanup climate from blocking stuff / migrate generic

* Migrate generic thermostat

* fix tests

* fix lint

pvizeli committed:
Cleanup media_player universal platform / blocking platform. (#5671)

* Cleanup media_player universal platform / blocking platform.

* fix comments

* fix unittest

* fix lint

* fix unittest lint

* fix coro

* fix test

* fix tests part 2

balloob committed:
Upgrade zeroconf dep" (#5706)

tboyce021 committed:
Fix OwnTracks state names (#5454)

* Fix OwnTracks state names (#5453)

* Update owntracks.py

* Update tests

Commit:
Wink AC and addidtional sensor support (#5670)

* Added door bell sensors

* Initial support for AC units.

* Added new device service

* Quirky Aros AC unit support

* Use super() everywhere and error checking for token request.

* Ignore camera sensors during setup of alarms.

* Added manufacturer/device attributes to all wink devices.

* Fixed style errors

* Fixed remaining lint errors.

benvm committed:
Modbus write_register accept list

pschmitt committed:
Support for Nuki.io smart locks (#5715)

* Support for Nuki.io smart locks

* Update requirements and add lock.nuki to .coveragerc

* lint: Re-organize imports

* Schedule a state update instead of calling directly update_ha_state

* Remove update requests altogether

* Make sure there is no IO inside properties

* Fix: nuki lock are all initialized as "lock.unnamed_device"

* Update pynuki to 1.2 to avoid an extra REST API call for each lock init

Duoxilian committed:
Cleanup climate and ecobee (#5616)

* Remove redundant input validation which is already accomplished through
defined schemata.

* Rely on defined state attributes for hold mode.

* Remove misleading comment. This comment seems to assume that sleep
mode is a hold; it is a schedule instead. The code snippets in the
comment could never work.

* Remove use of constants for hold mode. Will be made irrelevant
by a planned change by nordlead2005.

Danielhiversen committed:
up rfxtrx lib (#5721)

Danielhiversen committed:
update miflora lib and allow specification of bluetooth adapter (#5720)

pvizeli committed:
Migrate fan component to async. (#5723)

* Migrate fan component to async.

* Fix lint

colinodell committed:
QNAP Sensor (#5666)

* Implement the QNAP sensor

* Add sensors immediately

* Remove unnecessary check

* Use CONF_SSL instead of CONF_PROTOCOL

pvizeli committed:
Migrate cover to async. (#5717)

valentinalexeev committed:
Bump pwaqi to 1.4 to fix a typo in the underlying library. (#5716)

miniconfig committed:
Updated openvse sensor component to use new structure of openevsewifi library.

emlove committed:
[recorder] Index events time_fired to improve logbook performance (#5633)

* Index events time_fired to improve logbook perf.

* Updated implementation to track schema versions

* Added tests for schema migration support logic

* Rename check_schema to migrate_schema

miniconfig committed:
Merge pull request #5658 from miniconfig/openevse

Added new sensor component to monitor OpenEVSE chargers.

rmkraus committed:
Alert Component (#5201)

BastianPoe committed:
Add support for FRITZ!DECT wireless switches based on fritzhome (#5541)

wookiesh committed:
Adds average load to systemmonitor (#5686)

* Adds average load to systemmonitor

* split the values in 3 sensors

* hound ok

fabaff committed:
Add moon sensor (#5726)

* Add moon sensor

* Update moon.py

pschmitt committed:
Support for the Orange Livebox Play TV appliance (#5533)

* Support for the Orage Livebox Play TV appliance

* Add liveboxplaytv to coveragerc

* Minor refactoring

* Update requirements

* Adjust comments

* Fix alignment

* Fix some coding-style issues highlighted by Travis CI

* The livebox play TV does not support playing media

* Lint: shorten line

* Remove unused callback function

* Remove redundant backslash

* Implement changes requested by balloob

* Don't error out if channel name or media url could not be retrieved

* Support current program (media title property)

* Remove unnecessary check

* Clean up: Remove another unnecessary check, _CONFIGURING variable and _playing attribute

* Update liveboxplaytv dependency to version 1.4.4

* Fix liveboxplaytv requirement

* Improve media state (support for playing and pause state)

* Update liveboxplaytv.py

fabaff committed:
Upgrade pyowm to 2.6.1 (#5729)

fabaff committed:
Add wind bearing (#5730)

Commit:
Update python-wink version (#5734)

fabaff committed:
Upgrade psutil to 5.1.1 (#5736)

Joeboyc2 committed:
Change CONF_DEFAULT_COLOR CV type (#5700)

* Change CONF_DEFAULT_COLOR CV type

Changed  vol.Optional(CONF_DEFAULT_COLOR, default=DEFAULT_COLOR) from cv.string to cv.ensure_list

This allows the optional parameter default_color to be picked up correctly and to function

the option needs to be specifed as follows:

    default_color: [0,255,0]

Solution provided by @scossa2020 in issue #5338
https://github.com/home-assistant/home-assistant/issues/5338

* Update hyperion.py

fabaff committed:
Upgrade astral to 1.3.4 (#5744)

fabaff committed:
Upgrade psutil to 5.1.2 (#5745)

fabaff committed:
Upgrade zeroconf to 0.18.0 (#5746)

MartinHjelmare committed:
Fix validation of serial port on windows (#5749)

* Fix validation of serial port on windows

* Use pyserial to check serial ports.
* Check that persistence file ends with either `.json` or `.pickle`.

* Change fix to not rely on pyserial

* Use generator expr instead of list comprehension

fabaff committed:
Upgrade pyasn1 to 0.2.1 (#5755)

pschmitt committed:
Store the key file in the config dir (#5732)

* webostv: Store the key file in the config dir

* Update the pylgtv source to use the repo by @TheRealLink

* Add missing config parameter

fabaff committed:
Upgrade pysnmp to 4.3.3 (#5757)

pvizeli committed:
Add unittest for multible entities in one line (#5759)

MartinHjelmare committed:
Update mysensors sensors (#5764)

* Add V_DIRECTION type for S_WIND.
* Add more predefined units of measurement.

Yannic-HAW committed:
- added send packet service to send ir packets directly from scripts (no need to define a switch for each command) (#5768)

- changed ip suffix of SERVICE_LEARN to use _ instead of . as ip seperator to avoid errors if used in yaml

Danielhiversen committed:
flux led o.13 (#5774)

aequitas committed:
Make brightness display work for rgb devices. (#5675)

* Make brightness display work for rgb devices.

Non rgbw devices return 255 for getWarmWhite255. This is part 2 to make the brightness slider work for these devices.

https://github.com/Danielhiversen/flux_led/pull/25

* Query brightness property which return WW level or RGB brightness equivalent.

https://github.com/Danielhiversen/flux_led/pull/25

andrey-git committed:
Merge pull request #5718 from benvm/modbus_write_register_array

Modbus write_register doesn't accept list

adrianlzt committed:
Allow to use data for enhanced messages (#5763)

Add notification data field to the message send to Facebook.
Allows to construct richer messages like cards, quick replies, attach
images, videos, etc

albertoarias committed:
Add support for position on wink cover (#5751)

Add support for position property for wink covers and fix state when stopped in the middle.

pvizeli committed:
Migrate lock component to async (#5748)

andriej committed:
Update package.py (#5783)

tchellomello committed:
Fix attribute error for media_player/roku if roku device is unreachable and shows a persistent notification (#5785) (#5786)

ray0711 committed:
Fix brightness slider for mqtt template lights (#5780)

Add the missing supported_features declaration.

sbidoul committed:
add min and max jinja filters (#5765)

pvizeli committed:
Sonos fix favorite, coordinator, cleanup update (#5778)

* Sonos fix favorite, coordinator, cleanup update

* Bugfix snapshot restore

alexmogavero committed:
Improve Sky Hub error handling (#5762)

* Added error handling in function _get_skyhub_data

* Error line split for readability

Commit:
Added error checking to the MIMEImage encoding for smtp.py (#5753)

* Added error checking to the MIMEImage encoding for smtp.py

Added fallback to file attachment rather than inline image for images
without a known MIME

* PEP8 reqs to fix previous commit

postlund committed:
Initial support for Apple TV (#5698)

* Initial support for Apple TV

* hash_wip

* Add media_play support to Apple TV

herm committed:
Check for command topics when determinig the capabilities of an MQTT light. (#5770)

Previous code used the state topic which is obviously wrong:
- The state topic is already used to select optimistic mode.
- A light with only the state topic but no command topic would still announce the capability.

andrey-git committed:
Check config before restarting (#5609)

* Check config before restarting.

* Make check_config on restart async

* don't check if notification service exists

* Use .communicate()

* Reduce the number of notifications. Add tests.

tboyce021 committed:
Fix Hue groups with same names (#5737)

* Revert "Fix hue lightgroups not syncing state (#5702)"

* Use light_id in unique_id for Hue groups

* Make sure HueLight unique_id is unique

* Update hue.py

* Update hue.py

* Update hue.py

tboyce021 committed:
Add icon_template to template sensor (#5766)

* Add icon_template to template sensor

* Update test_template.py

* Update test_template.py again

* Update template.py

* Update test_template.py

* Update test_template.py

robbiet480 committed:
TTS ID3 support (#5773)

* Add support for writing ID3 tags to the file for improved display in media players

* Lint and async fixes

* Use mutagen instead of taglib

* Fix tests

* Add fallback for album

* Requested changes

* move import

* Fix album name

* Change default options handling

* Move to member function / minor fix

* fix style

* fix lint

* change mutagen handling

* fix lint / add name to bytesio

* Update __init__.py

* Fix test, some cleanups

* Add mutagen exeption handling, fix tests

* fix mutagen taging

fabaff committed:
MQTT discovery (#5724)

* Change implementation

* Re-write

* Remove unused consts

* Update discovery.py

* Add tests

* fix other tests

* Fix check_config script test

* Lint

* Lint

homeassistant commented:
Hi @kylehendricks, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#5792).

Once you do that we will be able to review and accept this pull request.

Thanks!

pschmitt commented:
573fc65 may not be worth its own section in the release notes but it contains a pseudo-breaking change for `media_player.webostv` users.  When upgrading users need to move `$HOME/.pylgtv` to `$HASS_CONFIG_DIR/webostv.conf` or Home Assistant will need to be paired with the TV again.

andrey-git commented:
## Custom state-card UI

This version adds the ability to use custom state-card UI
Go ahead and write your own state-card for [light](https://home-assistant.io/cookbook/custom_ui_by_andrey-git), sensor, lock, etc.
https://home-assistant.io/developers/frontend_creating_custom_ui/

MartinHjelmare committed:
Add mysensors device tracker and platform discovery (#5781)

* Add mysensors device_tracker platform

* Add discovery of device_tracker platforms

* Enable discovery of device_tracker platforms that are not
  DeviceScanner.
* Update signature of setup_scanner function in all affected platforms.
* Add test.
* Use discovery for mysensors device_tracker platform.

* Remove gps accuracy

* Small change to core like schema

* fix depency

homeassistant commented:
Hi @kylehendricks, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#5792).

Once you do that we will be able to review and accept this pull request.

Thanks!

pvizeli committed:
Revert last change on TTS cache load for more speed (#5797)

fabaff committed:
Upgrade pyasn1 to 0.2.2 (#5796)

JshWright committed:
Update python-nest dependency version (#5795)

The nest-cam changes have now been merged into the upstream library, so
there is no need to track a specific branch.

Updating to 3.1.0 also fixes a structure parsing bug I was experiencing.

robbiet480 committed:
Bump netdisco to 0.8.2

andrey-git committed:
Create a file for zwave workarounds. (#5798)

* Create a file for zwave workarounds. Add sensor->binary_sensor for fgfs101 (#2)

* Don't use default None

jawilson committed:
Fix climate.set_fan_mode yaml (#5799)

emlove committed:
Formalize supported_features as entity property (#5794)

* Formalize supported_features as entity property

* Remove extra emulated_hue conditions

* Generate log message in executor

balloob committed:
Update frontend (#5800)

kellerza committed:
WIP: [component/recorder] Refactoring & better handling of SQLAlchemy Sessions (#5607)

* Refactor recorder and Sessions

* Cover #4352

* NO_reset_on_return

* contextmanager

* coverage

aequitas committed:
Add support for fluxled discovery. (#5784)

* Add support for fluxled discovery.

* Make use of device type/protocol auto detection.

pschmitt committed:
Mailgun notify service (#5782)

* Mailgun notify service

* Update dependency to version 1.3

- The provided credentials (including the domain) are now checked during
startup, as requested by @balloob
- The domain name is now optional
- There's a new config item "sandbox" which indicates whether to use the
sandboxed domain in case the domain is not set

* Fix a few lint issues

* Disable lint check no-value-for-parameter

postlund committed:
Add discovery suppport to Apple TV (#5801)

Add an optional extended description…

nodinosaur committed:
The word `router` was misspelt (#5803)

fixed as this message is user facing

kellerza committed:
Update sma.py (#5807)

Danielhiversen committed:
Improve warning message in template rendering (#5806)

* improve warning message when template is none

* improve error message when template is none

* improve error message when template is none

* improve error message when template is none

pvizeli committed:
Move core service from core to components (#5787)

* Move core servcie from core to components

* add new handler for signals/exception

* Static persistent id

* Move unittest

* fix coro/callback

* Add more unittest for new services

* Address comments

* Update __init__.py

pvizeli committed:
[Image_Processing][Breaking Change] Cleanup Base face class add support for microsoft face detect (#5802)

* [Image_Processing] Cleanup Base face class add support for microsoft face detect

* fix lint

* add unittest for micosoft detect

* fix test

andrey-git committed:
Make sure workaround_component is not none (#5808)

pschmitt commented:
> The support for LG webOS Smart TVs was improved

@fabaff 
This is inaccurate. The only thing that has changed is the location of the configuration file holding the pairing key. No new feature has been introduced, nothing has been "improved".

I'd reword it like this:

### Breaking changes

- [LG webOS Smart TVs](https://home-assistant.io/components/media_player.webostv/) users: the location of the file holding the pairing key has changed. This requires you to move `$HOME/.pylgtv` to `$HASS_CONFIG_DIR/webostv.conf` or Home Assistant will need to be paired with the TV again.

rytilahti committed:
Add available property and typing hints (#5593)

* light.demo: add available property, add typing hints

* light.demo: keep all lights available, fix init ordering

* Fix issues raised during review

* Update demo.py

acambitsis committed:
Upgrade Russound integration to v0.1.7 (#5756)

* Bumped up version to use 0.1.7 of Russound integration module.
Fixed bug arising from not supporting TURN_ON state (fixes issue https://github.com/home-assistant/home-assistant/issues/5012)
Implemented state support in 0.1.7 such that component state is returned from the actual AMP. (Still uses polling model though).
Tested it with home-assitant users @laf (original developer of the module) and @hofsta.  Works fine with their Russounds.

* Made styling / compliance changes and updated correct version of russound module on requirements_all.txt.

* Changed handling of properties to be compliant with https://github.com/home-assistant/home-assistant/issues/4210
(Specifcailly added member variables for state, volume and source to cache these values, and introduced Update() method to set their values).

Now returns None if the selected source index that is returned from russound is greater than the length of the specified source list in the yaml config.
Removed unnecesary comment.

* Removed blank line after docstring.

* Removed updated() in class init and added True paramter to add_devices in setup_platform.

* Dropped the no longer needed self.update()

kellerza committed:
Initial (#5811)

pvizeli committed:
Ffmpeg update 1.4 (#5813)

* Pump ffmpeg version

* update entity

* next 1.4

Teagan42 committed:
Change medium state for filtering (#5817)

sdague committed:
Enhancements to ARWN platform (#5816)

* Fix arwn platform discover_sensors

The discover_sensors function can return either singletons or a list
of sensors. However the consumer was always expecting a list. This
fixes it to work in both cases.

* Add custom icons to arwn sensors.

This adds some custom icons for different kinds of weather sensors
that the arwn platform returns. Makes it a little easier to see what's
going on.

Commit:
Fixed call to object_id() (#5814)

kellerza committed:
[recorder] Protect against running in the event loop (#5812)

turbokongen committed:
Bugfixes (#5740)

* wrong data for lock alarm_type

* missing whitespace

* Not possible to set codes starting with 0

Duoxilian committed:
Support away_mode as permanent hold and hold_mode as temporary hold. (#5725)

* Support away_mode as permanent hold and hold_mode as temporary hold.

* Add comments to explain code better. Remove indefinite hold preference
to be consistent with 'away_mode'.

balloob committed:
Remove non-working webfont preload (#5819)

balloob committed:
Update frontend (#5820)

balloob committed:
Upgrade aiohttp to 1.3 (#5821)

pvizeli committed:
Move signal handling out of core to bootstrap (#5815)

* Move signal handling out of core to bootstrap

* Fix tests

balloob committed:
Deprecate event decorators (#5822)

balloob committed:
Make device sun light trigger async (#5823)

turbokongen committed:
Adding helper for get and set values (#5743)

* cleanup

* Update __init__.py

* Update __init__.py

* Update __init__.py

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

balloob commented:
Canceling this PR in favor of the PR from a release branch: https://github.com/home-assistant/home-assistant/pull/5833
